tests/PHPStan/Analyser/nsrt/bug-13272.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Bug13272;
+
+use function PHPStan\Testing\assertType;
+
+function foo(object $bar): void
+{
+	foreach (['qux', 'quux'] as $method) {
+		assertType("'quux'|'qux'", $method);
+
+		if (!method_exists($bar, $method)) {
+			throw new \Exception;
+		}
```

Comment:
I think the we could build a better type in a `TypeSpecifyingExtension` if the `TypeSpecifierContext` would contain information, whether the context is terminating.